### PR TITLE
Clean map

### PR DESCRIPTION
**Pull request by felipeZ**
## Changes
Clean and refactor the `autogen_mapping` script

### CODE REVIEW

File: scripts/xtp_autogen_mapping.in [hunk 1]
```diff
@@ -15,301 +15,454 @@
...
+nprocs {ncores}
+end
+
+! DFT {opt} {meta.functional} {meta.basis} {ri_string}
+! D3BJ
+{chelpg}
+
+% elprop
+polar 1
+end
```
[the remark this answers is not in the document]
Comment (felipeZ) [reply scripts/xtp_autogen_mapping.in:100]: I forgot to enable it only for the optimization, I am changing it!

**Review by baumeier — APPROVED** [review]

File: share/xtp/xml/molpol.xml [hunk 2]
```diff
@@ -2,6 +2,7 @@
 <options>
   <molpol help="Tool to scale distributed polarisibilites to DFT polarisibilites" section="sec:overlap">
     <options_polar help="Options for the electrostatic interaction">
+      <polar/>
```
[the remark this answers is not in the document]
Comment (felipeZ) [reply share/xtp/xml/molpol.xml:5]: I think that you only need the polar tag to be present

**Comment by felipeZ** [comment]
@baumeier I have changed two small things about the behaviour of the script:
1. the `-opt` or `--optimize` flag signals whether or not the optimization is perform without an extra arguments, for instance:
```bash
xtp_autogen_mapping -pdb CO.pdb -opt --orca $HOME/local/orca_4_2_1/orca -s n e h
```
Notice that the `-opt` option no longer takes an argument

2. I also print what command is currently running to let the user know, for instance:
```bash
xtp_autogen_mapping -pdb CO.pdb -opt --orca $HOME/local/orca_4_2_1/orca -s n e h
```
Produces the following output:
```bash
running:  /home/user/local/orca_4_2_1/orca CO_n.inp > CO_n.log 2> CO_n.err
running:  xtp_tools -e log2mps -n CO_n > CO_n_log2mps.log 2> CO_n_log2mps.err
running:  /home/user/local/orca_4_2_1/orca CO_h.inp > CO_h.log 2> CO_h.err
running:  xtp_tools -e log2mps -n CO_h > CO_h_log2mps.log 2> CO_h_log2mps.err
running:  /home/user/local/orca_4_2_1/orca CO_h_n.inp > CO_h_n.log 2> CO_h_n.err
running:  /home/user/local/orca_4_2_1/orca CO_e.inp > CO_e.log 2> CO_e.err
running:  xtp_tools -e log2mps -n CO_e > CO_e_log2mps.log 2> CO_e_log2mps.err
running:  /home/user/local/orca_4_2_1/orca CO_e_n.inp > CO_e_n.log 2> CO_e_n.err
```

**Comment by baumeier** [comment]
@felipeZ that looks good, nice job. Thx.

**Review by JensWehner — APPROVED** [review]
I like your python mojo.

Well done. Especially the types are really nice

**Comment by felipeZ** [comment]
> I like your python mojo.
> 
> Well done. Especially the types are really nice

Thanks!

**Pull request merged**